### PR TITLE
Smart Grid general improvements

### PR DESCRIPTION
**Pull request by ggallotti**
- Infinite scroll is now part of the Component and not of angular implementation. (in order to simplify angular declaration)
- Smart Grid: Fetch next page when reaches end. 
- Reduce the # of container updates. 
- Do not use mousewheel in order to scroll over horizontal grid.

### CODE REVIEW

File: src/components/grid-base/grid-base.tsx [hunk 1]
```diff
@@ -23,6 +23,11 @@ export interface IGridBase {
    */
   loadingState: "loading" | "loaded";
 
+  /**
+   * For infinite scroll, bind it to the next page grid component handler. It will be called every time threshold is reached.
+   */
+  gxInfinite: EventEmitter<void>;
```
Comment (ncardeli) [inline src/components/grid-base/grid-base.tsx]: Maybe we could give this event a more descriptive name? For example, `gxInfiniteThresholdReached`

File: src/components/grid-fs/readme.md [hunk 2]
```diff
@@ -1,24 +1,42 @@
 # gx-grid-fs
 
-Container to include repetitive element list. 
-It provides two slots:
-- 'grid-content' slot: Include the grid content here
-- 'grid-content-empty' slot: This slot will be shown, only if the grid is empty. 
+Container to include repetitive element list.
```
Comment (ncardeli) [inline src/components/grid-fs/readme.md]:
```suggestion
A container for including a repetitive elements list.
```

File: src/components/grid-fs/readme.md [hunk 3]
```diff
@@ -1,24 +1,42 @@
...
+It provides 3 slots:
 
-When grid is empty, a CSS Class name 'gx-grid-empty' is added to the host element. 
-When grid is loading, a CSS Class name 'gx-grid-loading' is added to the host element. 
-<!-- Auto Generated Below -->
+- 'grid-content' slot: Include the grid content here.
```
Comment (ncardeli) [inline src/components/grid-fs/readme.md]:
```suggestion
- `'grid-content'` slot: This slot holds the grid's content
```

File: src/components/grid-fs/readme.md [hunk 4]
```diff
@@ -1,24 +1,42 @@
...
+- 'grid-content-empty' slot: This slot will be shown, only if the grid is empty.
```
Comment (ncardeli) [inline src/components/grid-fs/readme.md]:
```suggestion
- 'grid-content-empty' slot: This slot will be shown only when the grid is empty.
```

File: src/components/grid-fs/readme.md [hunk 5]
```diff
@@ -1,24 +1,42 @@
...
+- 'grid-loading-content': Include the loading indicator markup here.
```
Comment (ncardeli) [inline src/components/grid-fs/readme.md]:
```suggestion
- 'grid-loading-content': This slot will be shown while the grid is being loaded. Include the loading indicator markup here.
```

File: src/components/grid-fs/readme.md [hunk 6]
```diff
@@ -1,24 +1,42 @@
...
+
+When grid is empty, a CSS Class name 'gx-grid-empty' is added to the host element.
```
Comment (ncardeli) [inline src/components/grid-fs/readme.md]:
```suggestion
When the grid is empty, a CSS Class named 'gx-grid-empty' is added to the host element.
```

File: src/components/grid-fs/readme.md [hunk 7]
```diff
@@ -1,24 +1,42 @@
...
+When grid is loading, a CSS Class name 'gx-grid-loading' is added to the host element.
```
Comment (ncardeli) [inline src/components/grid-fs/readme.md]:
```suggestion
When the grid is loading, a CSS Class named 'gx-grid-loading' is added to the host element.
```

File: src/components/grid-smart/grid-smart.tsx [hunk 8]
```diff
@@ -58,11 +57,21 @@ export class GridSmart
 
   @Prop() loadingState: "loading" | "loaded";
 
+  /**
+   * Logging level. For troubleshooting component update and initialization.
+   */
+  @Prop() logLevel: "debug" | "off" = "debug";
```
Comment (ncardeli) [inline src/components/grid-smart/grid-smart.tsx]: I'm not sure if log level should be part of the properties of the component. In any case, I'd rather have a global variable or a static member of a class to enable logging on all the components.

File: src/components/grid-smart/grid-smart.tsx [hunk 9]
```diff
@@ -359,15 +367,31 @@ export class GridSmart
...
+    return this.isInitialized();
+  }
+
+  private isInitialized(): boolean {
+    return this.swiper !== null;
+  }
+
+  private log(msg: any): void {
+    if (msg && this.logLevel !== "off") {
+      // console.log(msg, this.recordCount, this.loadingState, this.el);
```
Comment (ncardeli) [inline src/components/grid-smart/grid-smart.tsx]: I'm not sure about leaving this commented code here.

File: src/components/grid-smart/grid-smart.tsx (hunk 8 above)
Comment (ggallotti) [reply src/components/grid-smart/grid-smart.tsx]: Agree!
I'm working on small Logging framework for this in another branch.
So this code will be removed.

File: src/components/grid-base/grid-base.tsx (hunk 1 above)
Comment (ggallotti) [reply src/components/grid-base/grid-base.tsx]: Done!

File: src/components/grid-smart/grid-smart.tsx (hunk 9 above)
Comment (ggallotti) [reply src/components/grid-smart/grid-smart.tsx]: This will be resolved later. After introducing logging api.

**Pull request merged**